satip/eumetsat.py

```diff
@@ -432,27 +479,51 @@ def download_tailored_datasets(
                 parent="DownloadManager",
             )
             return
-
-        for dataset_id in dataset_ids:
-            # Download the raw data
-            try:
-                self._download_single_tailored_dataset(
-                    dataset_id,
-                    product_id=product_id,
-                    roi=roi,
-                    file_format=file_format,
-                    projection=projection,
-                )
-            except Exception:
-                log.debug("The EUMETSAT access token has been refreshed", parent="DownloadManager")
-                self.request_access_token()
-                self._download_single_tailored_dataset(
-                    dataset_id,
-                    product_id=product_id,
-                    roi=roi,
-                    file_format=file_format,
-                    projection=projection,
-                )
+        if parallel:
+            with ThreadPoolExecutor(max_workers=concurrency) as executor:
+                futures = [
+                    executor.submit(
+                        self.download_single_tailored_dataset_with_retry,
+                        dataset_id,
+                        product_id,
+                        roi,
+                        file_format,
+                        projection,
+                    ) for dataset_id in dataset_ids
+                ]
+
+                for future in as_completed(futures):
+                    try:
+                        future.result()
+                    except Exception as e:
+                        log.error(
+                            f"Failed to download dataset after retrying: {e}",
+                            parent="DownloadManager"
+                        )
+        else:
+            for dataset_id in dataset_ids:
```

Comment:
Actually, with thinking about it more, I would remove the non-parallel approach, and test. If the concurrency is defaulted to 1 instead, it still will test the futures work, and can be scaled up successfully, while cutting down on the test times.

Reply:
That makes sense! I'll implement that and test it out.